### ISSUE TITLE
damoyolo_tinynasL25_S_cigarette.pt convert onnx

### ISSUE THREAD

**Issue by mrlihellohorld** [tinyvision/DAMO-YOLO#167]
### Before Asking

- [x] I have read the [README](https://github.com/tinyvision/DAMO-YOLO/blob/master/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [ ] I want to train my custom dataset, and I have read the [tutorials for finetune on your data](https://github.com/tinyvision/DAMO-YOLO/blob/master/README.md#Finetune%20on%20your%20data) carefully and organize my dataset correctly; 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。

- [x] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [x] I have searched the DAMO-YOLO [issues](https://github.com/tinyvision/DAMO-YOLO/issues) and found no similar questions.

### Question

请问damoyolo微调的抽烟模型，如何转到onnx呢  目录结构是这样的

<img width="576" height="404" alt="Image" src="https://github.com/user-attachments/assets/64e2141f-170c-41b1-9c2c-4253ed02ee6f" />

**Comment by mrlihellohorld** [comment]
class DamoYoloONNXDetector:
    def __init__(self, config_path, onnx_model_path, device='cuda:0', infer_size=(640, 640), conf_threshold=0.5):
        self.config = parse_config(config_path)
        self.infer_size = infer_size
        self.conf_threshold = conf_threshold
        self.device = device if torch.cuda.is_available() and 'cuda' in device else 'cpu'
        self.class_names = self._get_class_names()

        # Load ONNX model
        self.onnx_session, self.input_name = self._load_onnx_model(onnx_model_path)

    def _get_class_names(self):
        if "class_names" in self.config.dataset:
            return self.config.dataset.class_names
        else:
            return [str(i) for i in range(self.config.model.head.num_classes)]

    def _load_onnx_model(self, onnx_path):
        providers = [('CUDAExecutionProvider', {'device_id': int(self.device.split(':')[-1])})
                     ] if 'cuda' in self.device else ['CPUExecutionProvider']
        session = ort.InferenceSession(onnx_path, providers=providers)
        input_name = session.get_inputs()[0].name
        return session, input_name

    def _pad_image(self, img_tensor, target_size):
        n, c, h, w = img_tensor.shape
        assert n == 1 and h <= target_size[0] and w <= target_size[1]
        padded = torch.zeros((n, c, target_size[0], target_size[1]))
        padded[:, :, :h, :w] = img_tensor
        return ImageList(padded, [img_tensor.shape[-2:]], [padded.shape[-2:]])

    def _preprocess(self, pil_img):
        np_img = np.array(pil_img.convert("RGB"))
        img = transform_img(np_img, 0,
                            **self.config.test.augment.transform,
                            infer_size=self.infer_size)
        padded = self._pad_image(img.tensors, self.infer_size)
        return padded, np_img.shape[:2]

    def _postprocess(self, output, image_size, origin_shape):
        scores = torch.tensor(output[0])
        bboxes = torch.tensor(output[1])
        result = postprocess(
            scores, bboxes,
            self.config.model.head.num_classes,
            self.config.model.head.nms_conf_thre,
            self.config.model.head.nms_iou_thre,
            image_size
        )
        result = result[0].resize(origin_shape)
        bboxes = result.bbox
        scores = result.get_field('scores')
        labels = result.get_field('labels')
        return bboxes, scores, labels

    def infer(self, pil_img):
        image_tensor, origin_shape = self._preprocess(pil_img)
        input_np = np.array(image_tensor.tensors.cpu())
        start_time = time.time()
        output = self.onnx_session.run(None, {self.input_name: input_np})
        inference_time = time.time() - start_time
        bboxes, scores, labels = self._postprocess(output, image_tensor, origin_shape)
        #print(bboxes, scores, labels)
        results = []
        for bbox, score, label in zip(bboxes, scores, labels):
            if score >= self.conf_threshold:
                results.append({
                    'bbox': bbox.tolist(),
                    'score': float(score),
                    'label': self.class_names[int(label.item())]
                })

        return results, inference_time

我转onnx进行推理，发现效果和使用modelscope进行推理效果差了很多

**Comment by mrlihellohorld** [comment]
请问有人能帮忙回答一下吗  谢谢
